### PR TITLE
Log write failures for watcher history document.

### PR DESCRIPTION
**Pull request by jakelandis**
The failure is correctly getting propagated, however, it is ignored.
This commit adds support to explicitly look for .watch-history failures
using the same logging strategy as triggered watch failures.

-------

This is pretty difficult to add a test (and the change is just an additional log message). 

To test manually...

add `"index.mapping.total_fields.limit" : "150"` to `watch-history-no-ilm.json` and/or `watch-history.json` in `x-pack/plugin/core/src/main/resources`

create a watch that will add a bunch of mappings

```
{
	"metadata" : {
		"a" : 0,
		"b" : 0,
		"c" : 0,
		"d" : 0,
		"e" : 0,
		"f" : 0,
		"g" : 0,
		"h" : 0,
		"i" : 0,
		"j" : 0,
		"k" : 0,
		"l" : 0,
		"m" : 0,
		"n" : 0,
		"o" : 0,
		"p" : 0,
		"q" : 0,
		"r" : 0,
		"s" : 0,
		"t" : 0,
		"u" : 0,
		"v" : 0,
		"w" : 0,
		"x" : 0,
		"y" : 0,
		"z" : 0,
		"aa" : 0,
		"ab" : 0,
		"ac" : 0,
		"ad" : 0,
		"ae" : 0,
		"af" : 0,
		"ag" : 0,
		"ah" : 0,
		"ai" : 0,
		"aj" : 0,
		"ak" : 0,
		"al" : 0,
		"am" : 0,
		"an" : 0,
		"ao" : 0,
		"ap" : 0,
		"aq" : 0,
		"ar" : 0,
		"as" : 0,
		"at" : 0,
		"au" : 0,
		"av" : 0,
		"aw" : 0,
		"ax" : 0,
		"ay" : 0,
		"az" : 0
	},
	"trigger": {
		"schedule": {
			"interval": "30s"
		}
	},
	"actions": {
		"my-logging-action": {
			"logging": {
				"level": "info",
				"category": "foo",
				"text": "Payload is {{ctx.payload}}"
			}
		}
	},
  "throttle_period_in_millis": 0
}
```

The error will now be displayed in log and will look like:
```
[2019-07-09T12:21:55,323][ERROR][o.e.x.w.Watcher          ] [node1] watch history could not be written [watch3_46a3f5a6-c925-454b-aaa7-d99d7b7f7644-2019-07-09T17:21:55.154236Z], failure [java.lang.IllegalArgumentException: Limit of total fields [150] in index [.watcher-history-10-2019.07.09] has been exceeded]
```

To see the mapping error from the origin, you can set
```
{
  "transient": {
    "logger.org.elasticsearch.action.admin.indices.mapping.put.TransportPutMappingAction": "debug"
  }
}
```

### CODE REVIEW

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java [hunk 1]
```diff
@@ -330,14 +330,22 @@ public void beforeBulk(long executionId, BulkRequest request) {
...
+                    Map<String, String> historyFailures = Arrays.stream(response.getItems())
+                        .filter(BulkItemResponse::isFailed)
+                        .filter(r -> r.getIndex().startsWith(HistoryStoreField.INDEX_PREFIX))
+                        .collect(Collectors.toMap(BulkItemResponse::getId, BulkItemResponse::getFailureMessage));
+                    if (triggeredFailures.isEmpty() == false) {
+                        String failure = triggeredFailures.values().stream().collect(Collectors.joining(", "));
                         logger.error("triggered watches could not be deleted {}, failure [{}]",
-                            triggeredWatches.keySet(), Strings.substring(failure, 0, 2000));
+                            triggeredFailures.keySet(), Strings.substring(failure, 0, 2000));
+                    } else if (historyFailures.isEmpty() == false) {
```
Comment (hub-cap) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java]: could there ever be a case where `triggeredFailures.isEmpty() == true` and `historyFailures.isEmpty() == true` ? Or is this if / else done on purpose.

**Review by hub-cap — APPROVED** [review]
LGTM just one question about the if / else

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java]: good call.  yes, it is possible for both errors to occur.
Will change from `if/else` to simply a second `if`

Comment (jakelandis) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java]: fixed in https://github.com/elastic/elasticsearch/pull/44129/commits/0d6b1ed5fd149f5910c43d8265ab33f5e50158bd

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Pull request merged**